UnityWeld_Editor/AnimatorParameterBindingEditor.cs

```diff
@@ -0,0 +1,337 @@
+using System;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.AnimatedValues;
+using UnityEngine;
+using UnityWeld.Binding;
+using UnityWeld.Binding.Internal;
+
+namespace UnityWeld_Editor
+{
+    [CustomEditor(typeof(AnimatorParameterBinding))]
+    public class AnimatorParameterBindingEditor : BaseBindingEditor
+    {
+        private AnimatorParameterBinding targetScript;
+
+        private AnimBool viewAdapterOptionsFade;
+
+        // Whether each property in the target differs from the prefab it uses.
+        private bool viewAdapterPrefabModified;
+        private bool viewAdapterOptionsPrefabModified;
+        private bool viewModelPropertyPrefabModified;
+        private bool viewPropertyPrefabModified;
+
+        private void OnEnable()
+        {
+            // Initialise reference to target script
+            targetScript = (AnimatorParameterBinding)target;
+
+            Type adapterType;
+
+            viewAdapterOptionsFade = new AnimBool(
+                ShouldShowAdapterOptions(targetScript.ViewAdapterTypeName, out adapterType)
+            );
+
+            viewAdapterOptionsFade.valueChanged.AddListener(Repaint);
+        }
+
+        private void OnDisable()
+        {
+            viewAdapterOptionsFade.valueChanged.RemoveListener(Repaint);
+        }
+
+        public override void OnInspectorGUI()
+        {
+            // Because of some of the work arounds done due to Unity's undocumented behaviour,
+            // When testing in the editor, some of this may run and can produce unexpected results
+            // So we disable this all while in play mode
+            if (EditorApplication.isPlaying)
+            {
+                EditorGUILayout.HelpBox("Cannot modify in play mode!", MessageType.Warning);
```

Comment:
Maybe we should disable the rest of the controls as well to make sure that users don't try to modify it while in play mode.

Reply:
What do you mean? I should add that to all of the binding editors?

Reply:
Hmm I guess this would apply to all the editors. If it's a problem with the others then probably don't worry about adding this, but I'll add a task to the roadmap Trello that we should either make it so you *can* edit in play mode, or all of the editors are disabled in play mode.